src/tests/page_heap_test.cc

```diff
@@ -64,7 +64,35 @@ static void TestPageHeap_Stats() {
   delete ph;
 }
 
+// Allocates all the last-level page tables we will need. Doing this before
+// calculating the base heap usage is necessary, because otherwise if any of
+// these are allocated during the main test it will throw the heap usage
+// calculations off and cause the test to fail.
+static void AllocateAllPageTables() {
+  // Make a separate PageHeap from the main test so the test can start without
+  // any pages in the lists.
+  tcmalloc::PageHeap* ph = new tcmalloc::PageHeap();
+  tcmalloc::Span *spans[20];
```

Comment:
Thanks for the patch. Can you add some comments about why 20 or 10? You want to ensure certain amount of metadata, then why we need several iterations? How can we ensure that test remains correct if we (or when we) change metadata data structures?

Reply:
It's a constant used in the actual tests down below. I pulled it out as a named constant to keep it in sync for the future instead of duplicating it more.

I'm doing the allocations in separate chunks to ensure the same metadata gets allocated as in the tests below by following the exact same pattern. I don't think that's guaranteed (especially for future changes) if I used fewer, larger chunks or something else.